mobile_wallet/src/lib.rs

```diff
@@ -349,6 +325,98 @@ fn sign_message_aux(input: &str) -> anyhow::Result<String> {
     ))?)
 }
 
+#[derive(common::SerdeDeserialize)]
+struct MicroCCDAmount {
+    #[serde(rename = "microGtuAmount")]
+    amount: u64,
+}
+
+#[derive(common::SerdeDeserialize)]
+struct AccountAddressJSON {
+    address: String,
+}
+
+#[derive(common::SerdeDeserialize)]
+#[serde(untagged)]
+enum JSONPayload {
+    UpdateContract {
+        amount:       MicroCCDAmount,
+        #[serde(rename = "contractAddress")]
+        address:      ContractAddress,
+        #[serde(rename = "receiveName")]
+        receive_name: String,
+        parameter:    String, // Hex encoded bytes
+        #[serde(rename = "maxContractExecutionEnergy")]
+        energy:       u64,
```

Comment:
We have `Energy` as a type in concordium-base crate, which is availble in this crate.